qemu/tests/pci_bridge.py

```diff
@@ -0,0 +1,311 @@
+import logging
+
+from virttest import error_context
+from virttest import env_process
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import utils_disk
+
+
+def prepare_pci_bridge(test, params, pci_bridge_num):
+    """
+    Prepare pci-bridge for guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param pci_bridge_num: VM pci-bridge number
+    """
+    if pci_bridge_num < 1:
+        test.cancel("There should be at least one pci-bridge!")
+
+    params["pci_controllers"] = ""
+    for index in range(pci_bridge_num):
+        params["pci_controllers"] += "pci_bridge%d " % index
+        params["type_pci_bridge%d" % index] = "pci-bridge"
+
+
+def prepare_images(test, params, image_num, pci_bridge_num, opr, device_num=0):
+    """
+    Prepare images attached to pci-bridge for guest
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param image_num: VM image number (except image1)
+    :param pci_bridge_num: VM pci-bridge number
+    :param opr: operation
+    :param device_num: Prepared device number
+    """
+    pci_bridges = params.objects("pci_controllers")
+
+    fmt_list = params.objects("disk_driver")
+    for i in range(image_num):
+        image = "stg%s" % i
+        params["images"] = ' '.join([params["images"], image])
+        params["image_name_%s" % image] = "images/%s" % image
+        params["image_size_%s" % image] = params["data_image_size"]
+        params["force_create_image_%s" % image] = "yes"
+        params["remove_image_%s" % image] = "yes"
+        params["blk_extra_params_%s" % image] = "serial=TARGET_DISK%s" % i
+        if opr != "hotplug_unplug":
+            if i >= len(fmt_list):
+                params["drive_format_%s" % image] = "virtio"
+            else:
+                params["drive_format_%s" % image] = fmt_list[i]
+            d_format = params["drive_format_%s" % image]
+            if d_format == "scsi-hd" and params["drive_format"] == "scsi-hd":
+                params["drive_bus_%s" % image] = 1
+            if d_format == "usb2":
+                params["usbs"] += " ehci"
+                params["usb_type_ehci"] = "usb-ehci"
+                content = "usbc_pci_bus_ehci"
```

Comment:
@ybduan  Still test usb device with `disk_driver = "virtio scsi-hd usb2"` ? 
And @PaulYuuu commented before: usb controller `usb-ehci` is not available for ppc.
```
USB devices:
name "nec-usb-xhci", bus PCI
name "pci-ohci", bus PCI, desc "Apple USB Controller"
name "qemu-xhci", bus PCI
```

Reply:
I still haven't go through the usb related code.  It is on going. 
@vivianQizhu Please also help have a look, i remember you work on it before.